src/pages/gallery/gallery.tsx

```diff
@@ -8,56 +8,46 @@ import "./gallery.css";
 
 type SortOrder = "chronological" | "ascending" | "descending";
 type GalleryGameMetadata = GameMetadata & { show: boolean };
+type Filter = {
+	query: string,
+	tags: string[],
+	sort: SortOrder
+};
 export default function Gallery({ games, tags }: { games: GameMetadata[], tags: string[] }) {
 	const [gamesState, setGamesState] = useState<GalleryGameMetadata[]>([]);
-	const [sortOrder, setSortOrder] = useState<string>("");
-	const [tagFilter, setTagFilter] = useState<string>("");
-	const [searchQuery, setSearchQuery] = useState<string>("");
+	const [filter, setFilter] = useState<Filter>({query: "", sort: "chronological", tags: [] })
 
-	function sortGames(games: GameMetadata[], order: SortOrder): void {
+	useEffect(() => {
+		const lowerCaseQuery = filter.query.toLowerCase();
+		const _games = games.filter(
+				game => game.title.toLowerCase().includes(lowerCaseQuery) || 
+				game.author.toLowerCase().includes(lowerCaseQuery)
+			) // filter by query
+			.filter(game => { // filter by tags
+				for (const tag of filter.tags) {
+					if (game.tags.indexOf(tag) === -1) return false;
+				}
+				return true;
+			});
+		setGamesState(sortGames(_games, filter.sort).map(game => ({ ...game, show: true })) as GalleryGameMetadata[]);
+
+	}, [filter]);
+
+	function sortGames(games: GameMetadata[], order: SortOrder): GameMetadata[] {
+		const _games = [...games];
 		if (order === "chronological") {
-			games.sort((a, b) => Date.parse(b.addedOn) - Date.parse(a.addedOn))
+			_games.sort((a, b) => Date.parse(b.addedOn) - Date.parse(a.addedOn))
 				.slice(0, 10)
 				.forEach(game => (game.isNew = true));
```

Comment:
Mutation of objects via forEach is often considered an antipattern - often better to map into new obj's entirely